lib/pluginmanager/xpack/local_installer.rb

```diff
@@ -0,0 +1,110 @@
+# encoding: utf-8
+require "pluginmanager/ui"
+require "pluginmanager/bundler/logstash_injector"
+require "pluginmanager/gem_installer"
+require "pluginmanager/errors"
+require "pluginmanager/xpack/pack"
+require "bootstrap/util/compress"
+require "rubygems/indexer"
+
+module LogStash module PluginManager module Xpack
+  class LocalInstaller
+    PACK_EXTENSION = ".zip"
+    GEMS_DIR = "gems"
+
+    attr_reader :local_file
+
+    def initialize(local_file)
+      @local_file = local_file
+    end
+
+    def execute
+      raise PluginManager::FileNotFoundError, "Cant file local file #{local_file}" unless ::File.exist?(local_file)
+      raise PluginManager::InvalidPackError, "Invalid format, the pack must be in zip format" unless valid_format?(local_file)
+
+      PluginManager.ui.info("Installing file: #{local_file}")
+      uncompressed_path = uncompress(local_file)
+      PluginManager.ui.debug("Pack uncompressed to #{uncompressed_path}")
+      pack = LogStash::PluginManager::Xpack::Pack.new(uncompressed_path)
+      raise PluginManager::InvalidPackError, "The pack must contains at least one plugin" unless pack.valid?
+
+      local_source = move_to_local_source(uncompressed_path)
```

Comment:
Since the zip doesnt ship with a manifest file we use the directory structure to know what need to be added to the gemfile (root gem vs dependencies gems)

The `move_to_local_source` is needed because the Gem::Indexer requires that special directory structure to genere the index files.